lib/x/errors/too_many_requests.rb

```diff
@@ -18,6 +18,33 @@ def reset_in
       [(reset_at - Time.now).ceil, 0].max
     end
 
-    alias_method :retry_after, :reset_in
+    def all_limits
+      @all_limits ||= limits.to_h do |limit|
+        [
+          limit,
+          {
+            "limit" => response["x-#{limit}-limit"].to_i,
+            "remaining" => response["x-#{limit}-remaining"].to_i,
+            "reset_at" => Time.at(response["x-#{limit}-reset"].to_i)
+          }
+        ]
+      end
+    end
+
+    def next_available_at
+      all_limits.values.select { |v| (v["remaining"]).zero? }.map { |v| v["reset_at"] }.max
+    end
+
+    def next_available_in
+      [(next_available_at - Time.now).ceil, 0].max
+    end
+
+    alias_method :retry_after, :next_available_in
+
+    private
+
+    def limits
+      @limits ||= response.to_hash.keys.filter_map { |k, _v| k.match(/x-(.*-limit.*)-limit/)&.captures&.first }
```

Comment:
I believe `_v` should removed, since you call `keys` before `filter_map`, so there are only keys (no values) mapped from the hash.

Reply:
Right, I changed that.